src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +265,39 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Check if ICE candidates are available in the queue and add the candidate
+        while(!pendingIceCandidatesQueue.isEmpty()) {
```

Comment:
This will turn into an infinite loop as soon as you have an ice candidate whose client id doesn't match as you are not removing the message

Reply:
I have updated the data structure to use a map of client ID and queue of pending ICE candidates. The item from the queue is purged after the `addIceCandidate` call, so there would not be any infinite loop. Even without this logic, there would not have been an infinite loop because I remove the candidate even if there isnt a match. Nevertheless, there was a bug in this logic and changing the data structure fixed the bug.